srcs/vm/cursors.c

```diff
@@ -38,35 +38,59 @@ t_cursor    *cursor_clone(t_cursor *origin)
 	return (cursor);
 }
 
-t_cursor *cursor_add(t_cursor **acursor, t_cursor *new)
+// TODO: Should place cursor in the beginning of the list
+t_cursor *cursor_add(t_arena *arena, t_cursor *clone)
 {
-	if (acursor != NULL && new != NULL)
-	{
-		new->next = *acursor;
-		*acursor = new;
-	}
-	return (*acursor);
+    t_cursor *cursor;
+
+    cursor = (t_cursor *)ft_memalloc(sizeof(t_cursor));
+
+    if (cursor == NULL)
+    {
+        ;// ERROR MEMORY
+    }
+    else
+    {
+        if (clone != NULL)
+            ft_memcpy(cursor, clone, sizeof(t_cursor));
+        cursor->id = arena->cursor_count;
+        cursor->next = NULL;
+        arena->cursor_count++;
+        if (arena->cursors == NULL)
+            arena->cursors = cursor;
+        else
+        {
+            cursor->next = arena->cursors;
+            arena->cursors = cursor;
+        }
+    }
+    return (arena->cursors);
 }
 
-t_cursor	*cursor_del(t_cursor *current)
+void    cursor_del(t_cursor **head, int id)
 {
+	t_cursor	*temp;
 	t_cursor	*prev;
-	t_cursor	*del;
 
-	del = current;
-	current = current->next;
-	if (del->prev == NULL)
-	{
-		free(del);
-		current->prev = NULL;
-	}
-	else
+    temp = *head;
+
+	if (temp != NULL && temp->id != id)
 	{
-		prev = del->prev;
-		free(del);
-		current->prev = prev;
+		*head = temp->next;
+		free(temp);
 	}
-	return (current);
+
+    while (temp != NULL && temp->id != id) 
```

Comment:
Use after free.. please fix

Reply:
what?

Reply:
You free temp 3 lines up, then do not set it to follow another pointer, so you are using freed memory there..

Reply:
You see that return statement there? I free temp ONLY if head of the list already is the item id we want to remove. so then we set ->next as the new head and we free temp.